tests/data/test_maestro.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+# encoding: utf-8
+#
+# Copyright 2022 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import numpy as np
+import os
+import pathlib
+import wave
+
+from mido import MidiFile, MidiTrack, Message
+from typing import List
+
+import apache_beam as beam
+from apache_beam.testing.test_pipeline import TestPipeline
+
+from basic_pitch.data.datasets.maestro import (
+    MaestroToTfExample,
+    MaestroInvalidTracks,
+    create_input_data,
+)
+from basic_pitch.data.pipeline import WriteBatchToTfRecord
+
+RESOURCES_PATH = pathlib.Path(__file__).parent.parent / "resources"
+MAESTRO_TEST_DATA_PATH = RESOURCES_PATH / "data" / "maestro"
+
+TRAIN_TRACK_ID = "2004/MIDI-Unprocessed_SMF_05_R1_2004_01_ORIG_MID--AUDIO_05_R1_2004_03_Track03_wav"
+VALID_TRACK_ID = "2004/MIDI-Unprocessed_SMF_02_R1_2004_01-05_ORIG_MID--AUDIO_02_R1_2004_06_Track06_wav"
+TEST_TRACK_ID = "2004/MIDI-Unprocessed_SMF_02_R1_2004_01-05_ORIG_MID--AUDIO_02_R1_2004_08_Track08_wav"
+GT_15M_TRACK_ID = "2004/MIDI-Unprocessed_SMF_02_R1_2004_01-05_ORIG_MID--AUDIO_02_R1_2004_05_Track05_wav"
+
+
+def create_mock_wav(output_fpath: str, duration_min: int) -> None:
```

Comment:
This and the other mocked file should be a pytest-fixture and if possible, please use pytest's tmpdir fixture as input. Also I see you're using this for 3 and 16 minute files. Do you need both times? If not, then you can remove duration_min.

Reply:
can you use pytest-fixtures in non test methods? I am using it in the tests, if you look, just not directly as an argument to the method. Additionally because there's a bit of file organization going on within the tmpdir.

Reply:
carrying from offline chat - It seems like you can use fixtures with helper methods, but the function call from within test_methods gets a bit weird if you want to call other arguments. If I were to get rid of the output_fpath argument and make all files, I would still want to include a filename argument since there are tests that have multiple different files created. The time argument is also useful, since in one test I want files under 15m and in the other, over 15m. I suppose since i'm only using two different values, I could have a boolean value? But i'm not sure what marginal utility that adds.